PathOfBuildingCommunity/PathOfBuilding
# Fix trigger rate for Kitava's Thrist and Craft. Improve handling of cooldown overrides and multiple trigger scenarios.

Fixes #2444 , #2592.

### Description of the problems being solved:
Correct calculation of triggers for Kitava's Thirst and Craft trigger. Change Kitava's Thirst to use the in game trigger mod and add parsing for it. Improved handling of hard skill cooldown overrides. Improved handling of a gem being supported by more than one trigger.

### Steps taken to verify a working solution:
- Check specific cases with hand crafted items
- Check around a dozen builds from poe.ninja
- Read wiki and test in game

### Before screenshots:
Wave of Conviction:
![kitavawave_before](https://user-images.githubusercontent.com/91493239/178469959-a859135d-e7fb-431f-bf3d-1ec0d2ba28eb.jpg)

Frost Bomb:
![kitavawave_frost_before](https://user-images.githubusercontent.com/91493239/178469986-192c719a-5b89-4a5c-a25f-2b3ab49dde28.jpg)

Discharge with cooldown override jewel:
![kitavawave_discharge_before](https://user-images.githubusercontent.com/91493239/178470029-a440b43e-98c7-44e5-b165-2f0732deda71.jpg)

### After screenshots:
Wave of Conviction:
![kitavawave_after](https://user-images.githubusercontent.com/91493239/178470118-41a7dd7e-ae64-4d39-8e1e-c61bf9d03ca9.jpg)

Frost Bomb:
![kitavawave_frost_after](https://user-images.githubusercontent.com/91493239/178470143-04e65254-617d-45d8-8ede-82106d8069a0.jpg)

Discharge with cooldown override jewel:
![kitavawave_discharge_after](https://user-images.githubusercontent.com/91493239/178470172-be3deba0-b1a3-4b37-ad97-21218e53110b.jpg)

Skill disable when supported by more than one trigger:
![Skill_disable_multiple_trigger](https://user-images.githubusercontent.com/91493239/178470190-b283dd07-2f5a-4e5f-ae94-7ffccb36911c.jpg)
